### PR TITLE
chore: log KubeSpan IPs overlaps

### PR DESCRIPTION
**Pull request by AlekSi**
Should be helpful for debugging.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4485)

### CODE REVIEW

File: internal/app/machined/pkg/controllers/kubespan/peer_spec.go [hunk 1]
```diff
@@ -127,9 +127,22 @@ func (ctrl *PeerSpecController) Run(ctx context.Context, r controller.Runtime, l
 						continue
 					}
 
+					ipSetPrefixes := ipSet.Prefixes()
```
Comment (smira) [inline internal/app/machined/pkg/controllers/kubespan/peer_spec.go:130]: should we extract a function which stringifies `IPSet` and use it with logging?

as I don't think we need to build this list if there're no overlaps (and we can build it only when logging)

Comment (AlekSi) [reply internal/app/machined/pkg/controllers/kubespan/peer_spec.go:130]: Sounds good.

Do we want to log that by default? Or only in debug builds?

**Review by Unix4ever — APPROVED** [review]

File: internal/app/machined/pkg/controllers/kubespan/peer_spec.go (hunk 1 above)
Comment (smira) [reply internal/app/machined/pkg/controllers/kubespan/peer_spec.go]: I think by default is good!

**Review by smira — APPROVED** [review]

**Pull request merged**